### PR TITLE
[AArch64] Implement lmw/stmw.

### PR DESCRIPTION
**Pull request by Sonicadvance1**

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp [hunk 1]
```diff
@@ -515,3 +515,96 @@ void JitArm64::stX(UGeckoInstruction inst)
...
+
+	u32 a = inst.RA;
+
+	ARM64Reg WA = gpr.GetReg();
+	ARM64Reg XA = EncodeRegTo64(WA);
+	if (a)
+	{
+		bool add = inst.SIMM_16 >= 0;
+		u16 off = std::abs(inst.SIMM_16);
+		if (off < 4096)
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp]: should brace this external if as well.

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp [hunk 2]
```diff
@@ -515,3 +515,96 @@ void JitArm64::stX(UGeckoInstruction inst)
...
+			ARM64Reg WB = gpr.GetReg();
+			MOVI2R(WB, off);
+			if (add)
+				ADD(WA, gpr.R(a), WB);
+			else
+				SUB(WA, gpr.R(a), WB);
+			gpr.Unlock(WB);
+		}
+	}
+	else
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp]: and this else.

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp [hunk 3]
```diff
@@ -515,3 +515,96 @@ void JitArm64::stX(UGeckoInstruction inst)
...
+
+	ARM64Reg WA = gpr.GetReg();
+	ARM64Reg XA = EncodeRegTo64(WA);
+	ARM64Reg WB = gpr.GetReg();
+
+	if (a)
+	{
+		bool add = inst.SIMM_16 >= 0;
+		u16 off = std::abs(inst.SIMM_16);
+		if (off < 4096)
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp]: braces

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp [hunk 4]
```diff
@@ -515,3 +515,96 @@ void JitArm64::stX(UGeckoInstruction inst)
...
+		{
+			MOVI2R(WB, off);
+			if (add)
+				ADD(WA, gpr.R(a), WB);
+			else
+				SUB(WA, gpr.R(a), WB);
+		}
+	}
+	else
+		MOVI2R(WA, (u32)(s32)(s16)inst.SIMM_16);
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp]: here too.

File: Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp (hunk 1 above)
Comment (Sonicadvance1) [inline Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStore.cpp]: 4096 is the maximum size encoding in ADD/SUB without shifting the immediate by 12bits.

**Pull request merged**